yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/ActivityDetailsFactory.java

```diff
@@ -40,11 +46,25 @@ ActivityDetails create(Receipt receipt, PrivateKey privateKey) throws ProfileExc
 
         Profile userProfile = profileReader.read(receipt.getOtherPartyProfile(), secretKey);
         Profile applicationProfile = profileReader.read(receipt.getProfile(), secretKey);
+
+        ExtraData extraData = parseExtraData(receipt.getExtraData());
+
         String rememberMeId = parseRememberMeId(receipt.getRememberMeId());
         String parentRememberMeId = parseRememberMeId(receipt.getParentRememberMeId());
         Date timestamp = parseTimestamp(receipt.getTimestamp());
 
-        return new SimpleActivityDetails(rememberMeId, parentRememberMeId, userProfile, applicationProfile, timestamp, receipt.getReceiptId());
+        return new SimpleActivityDetails(rememberMeId, parentRememberMeId, userProfile, applicationProfile, extraData, timestamp, receipt.getReceiptId());
+    }
+
+    private ExtraData parseExtraData(byte[] extraDataBytes) throws ProfileException {
```

Comment:
I know we have ProfileExceptions being thrown elsewhere in ActivityDetails, but I don't think it makes sense to throw a ProfileException here, since ExtraData is not part of the profile, is there a suitable alternative?

Reply:
As we talked about, in the next major we are going to review the use of checked exceptions so we decided to wrap the ExtraDataException into a profile exception to not make it a breaking change for users.